app/forms/webauthn_setup_form.rb

```diff
@@ -92,6 +101,21 @@ def safe_response(original_origin)
     false
   end
 
+  def process_authenticator_data_value(data_value)
+    return unless data_value.present?
+    # Ensure that its an int
+    return unless data_value.scan(/\D/).empty?
+    data_int_value = data_value.to_i
+    {
+      up: ActiveModel::Type::Boolean.new.cast(data_int_value & (1 << 0)),
```

Comment:
Interesting, I was wondering why we use `ActiveModel::Type::Boolean.new.cast` here, though I guess it's because it [handles](https://api.rubyonrails.org/classes/ActiveModel/Type/Boolean.html) `0` as `false` in the way we want it to, unlike default Ruby boolean coercion e.g. `!!0 == true` ?

Also wonder if `> 0` would suffice, though perhaps this is clearer. No strong preference from me.

Reply:
+1 I think `> 0` or `().positive?` might be easier on the eyes, maybe we drop a comment about it being a boolean cast